include/view-impl-common.h

```diff
@@ -20,4 +20,9 @@ void view_impl_map(struct view *view);
  */
 void view_impl_apply_geometry(struct view *view, int w, int h);
 
+void view_init(struct view *view);
+void view_destroy(struct view *view);
```

Comment:
- rebased
- moved `view_init()` and `view_destroy()` to `src/view-impl-common.c` (which required to add a bunch of new includes there)
- reduced the additional `struct view_impl` function pointers to 2: `listeners_init()` and `listeners_remove()`. I decided against using `listeners_finish()` for now because the term `finished` really can mean a buch of things and `remove` is pretty clear in this case. I'd could change it to `listeners_finish()` though if it is really necessary (but would prefer not to).

Should we rename `view_init()` / `view_destroy()` to `view_impl_init()` / `view_impl_destroy()` now that they have been moved? They are only called from `src/xdg.c` and `src/xwayland.c`.

Reply:
Looks good. 

Yes, `view_impl_{init,destroy}` is better.